src/routes/TraderForm/Form/MarginAmountField/index.tsx

```diff
@@ -1,39 +1,46 @@
 import { TokenField, TokenFieldProps } from 'brokoli-ui';
-import React, { useCallback } from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 
 import {
   selectInfoPostSwap,
   selectIsMarginRequiredError,
   selectIsWalletMarginError,
-  selectMarginAmount,
   selectSwapFormAMM,
+  selectUserInputMarginInfo,
   selectWalletBalanceInfo,
   setMarginAmountAction,
   SwapFormNumberLimits,
 } from '../../../../app/features/swap-form';
 import { useAppDispatch, useAppSelector } from '../../../../app/hooks';
-import { compactFormat, formatNumber } from '../../../../utilities/number';
+import { compactFormat, formatNumber, stringToBigFloat } from '../../../../utilities/number';
 import { MarginAmountFieldBox } from './MarginAmountField.styled';
 type NotionalAmountProps = {};
 
 export const MarginAmountField: React.FunctionComponent<NotionalAmountProps> = () => {
   const dispatch = useAppDispatch();
-  const marginAmount = useAppSelector(selectMarginAmount);
+  const marginAmount = useAppSelector(selectUserInputMarginInfo);
   const infoPostSwap = useAppSelector(selectInfoPostSwap);
   const aMM = useAppSelector(selectSwapFormAMM);
 
   const isMarginRequiredError = useAppSelector(selectIsMarginRequiredError);
   const isWalletMarginError = useAppSelector(selectIsWalletMarginError);
 
+  const [localMargin, setLocalMargin] = useState<string | null>(marginAmount.value.toString());
+
   const walletBalance = useAppSelector(selectWalletBalanceInfo);
+
+  useEffect(() => {
+    setLocalMargin(marginAmount.value.toString());
+  }, [marginAmount.value]);
+
   const handleOnChange = useCallback(
     (value?: string) => {
-      if (!value) {
-        return;
-      }
+      setLocalMargin(value ?? null);
```

Comment:
the point of having a localMargin is because we want to debounce the event for dispatching that updates our global store. In this case you are missing the part for debouncing the dispatch. Can you add that as well? And move the dispatch to this debounced function

Reply:
[161562c](https://github.com/Voltz-Protocol/voltz-ui/pull/582/commits/161562c53ef4a56fb707817c0e17c85adcfc6ae0)